blueocean-dashboard/src/main/js/creation/credentials/CredentialsApi.js

```diff
@@ -0,0 +1,102 @@
+import { capabilityAugmenter, Fetch, UrlConfig } from '@jenkins-cd/blueocean-core-js';
+import TempUtils from '../TempUtils';
+
+export class CredentialsApi {
+
+    constructor(fetch) {
+        this._fetch = fetch || Fetch.fetchJSON;
+    }
+
+    listAllCredentials() {
+        const path = UrlConfig.getJenkinsRootURL();
+        const searchUrl = TempUtils.cleanSlashes(`${path}/blue/rest/search?q=type:credential`, false);
+
+        return this._fetch(searchUrl)
+            .then(data => capabilityAugmenter.augmentCapabilities(data));
+    }
+
+    saveUsernamePasswordCredential(username, password) {
+        const path = UrlConfig.getJenkinsRootURL();
+        const requestUrl = TempUtils.cleanSlashes(`${path}/blue/rest/organizations/jenkins/credentials/system/domains/_/credentials/`);
+
+        const requestBody = {
+            credentials: {
+                $class: 'com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl',
+                'stapler-class': 'com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl',
+                scope: null,
+                username,
+                password,
+                description: null,
+            },
+        };
+
+        const fetchOptions = {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(requestBody),
+        };
+
+        return this._fetch(requestUrl, { fetchOptions });
+    }
+
+    saveSshKeyCredential(privateKey) {
```

Comment:
I'll change it around. I know I'll get hazed for saying it was a convention I adopted after reading some Spring source code.  *walks away shamefully*